### PR TITLE
`azurerm_monitor_log_profile` - mark as deprecated

### PR DESCRIPTION
**Pull request by teowa**
from https://learn.microsoft.com/en-us/azure/azure-monitor/essentials/activity-log?tabs=powershell#legacy-collection-methods

On 30 September 2026, the legacy solution for forwarding Azure activity logs, known as Log Profiles, will be retired and replaced with diagnostic settings. This transition to diagnostic settings will happen automatically—you don’t need to take any action, and the change won’t cause any disruption in your workflow.

### CODE REVIEW

File: internal/services/monitor/monitor_log_profile_resource.go [hunk 1]
```diff
@@ -32,6 +32,8 @@ func resourceMonitorLogProfile() *pluginsdk.Resource {
 		Update: resourceLogProfileCreateUpdate,
 		Delete: resourceLogProfileDelete,
 
+		DeprecationMessage: "Azure Log profiles will be retired on 30th September 2026, for more details, please see https://learn.microsoft.com/en-us/azure/azure-monitor/essentials/activity-log?tabs=powershell#legacy-collection-methods",
```
Comment (tombuildsstuff) [inline internal/services/monitor/monitor_log_profile_resource.go:35]:
```suggestion
		DeprecationMessage: "Azure Log Profiles will be retired on 30th September 2026 and will be removed in v4.0 of the AzureRM Provider. More information on the deprecation can be found at https://learn.microsoft.com/azure/azure-monitor/essentials/activity-log?tabs=powershell#legacy-collection-methods",
```

File: website/docs/d/monitor_log_profile.html.markdown [hunk 2]
```diff
@@ -10,6 +10,8 @@ description: |-
 
 Use this data source to access the properties of a Log Profile.
 
+-> **NOTE:** Azure Log profiles will be retired on 30th September 2026, for more details, please see the [doc](https://learn.microsoft.com/en-us/azure/azure-monitor/essentials/activity-log?tabs=powershell#legacy-collection-methods).
```
Comment (tombuildsstuff) [inline website/docs/d/monitor_log_profile.html.markdown:13]: We should mark this as a red block:

```suggestion
!> **NOTE:** Azure Log Profiles will be retired on 30th September 2026 and will be removed in v4.0 of the AzureRM Provider. More information on the deprecation can be found [in the Azure documentation](https://learn.microsoft.com/azure/azure-monitor/essentials/activity-log?tabs=powershell#legacy-collection-methods).
```

File: website/docs/r/monitor_log_profile.html.markdown [hunk 3]
```diff
@@ -12,6 +12,8 @@ Manages a [Log Profile](https://docs.microsoft.com/azure/monitoring-and-diagnost
 
 -> **NOTE:** It's only possible to configure one Log Profile per Subscription. If you are trying to create more than one Log Profile, an error with `StatusCode=409` will occur.
 
+-> **NOTE:** Azure Log profiles will be retired on 30th September 2026, for more details, please see the [doc](https://learn.microsoft.com/en-us/azure/azure-monitor/essentials/activity-log?tabs=powershell#legacy-collection-methods).
```
Comment (tombuildsstuff) [inline website/docs/r/monitor_log_profile.html.markdown:15]: !> **NOTE:** Azure Log Profiles will be retired on 30th September 2026 and will be removed in v4.0 of the AzureRM Provider. More information on the deprecation can be found [in the Azure documentation](https://learn.microsoft.com/azure/azure-monitor/essentials/activity-log?tabs=powershell#legacy-collection-methods).

File: internal/services/monitor/monitor_log_profile_resource.go (hunk 1 above)
Comment (tombuildsstuff) [inline internal/services/monitor/monitor_log_profile_resource.go:35]: we should also mark the associated Data Source as deprecated too

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
hi @teowa 

Thanks for this PR - I've taken a look through and left some comments inline, but if we can fix those up then this otherwise should be good to go 👍 

Thanks!

**Review by stephybun — APPROVED** [review]
Thanks @teowa LGTM 👍

**Pull request merged**